lib/improver/wind_direction.py

```diff
@@ -333,34 +335,51 @@ def wind_dir_decider(self, where_low_r, first_member):
                 Array of boolean values. True where original wind direction
                 estimate has low confidence. These points are replaced
                 according to self.backup_method
-            first_member (np.array):
-                Array of wind direction data from the first ensemble
+            first_member (iris.cube.Cube):
+                Contains array of wind direction data from the first ensemble
                 realization
 
         Uses:
-            self.wdir_slice.data (np.ndarray):
-                3D array - wind direction angles from ensembles (in degrees).
-            self.wdir_slice_mean.data (np.ndarray):
-                2D array - average wind direction angle (in degrees).
+            self.wdir_slice_mean (iris.cube.Cube):
+                Containing average wind direction angle (in degrees).
+            self.wdir_complex (np.ndarray):
+                3D array - wind direction angles from ensembles (in complex).
             self.r_vals_slice.data (np.ndarray):
                 2D array - Radius taken from average complex wind direction
                 angle.
             self.r_thresh (float):
                 Any r value below threshold is regarded as meaningless.
+            self.realization_axis (int):
+                Axis to collapse over.
+            self.n_realizations (int):
+                Number of realizations available in the plugin. Used to set the
+                neighbourhood radius as this is used to adjust the radius again
+                in the neighbourhooding plugin.
 
         Defines:
             self.wdir_slice_mean.data (np.ndarray):
                 2D array - Wind direction degrees where ambigious values have
                 been replaced with data from first ensemble realization.
         """
-
         if self.backup_method == 'neighbourhood':
-            improved_values = np.full_like(self.wdir_slice_mean.data, None)
+            # Performs smoothing over a 6km square neighbourhood.
+            # Then calculates the mean wind direction.
+            nb_radius = math.sqrt((6000.**2.) * self.n_realizations)
```

Comment:
I'm not happy with this line.  Firstly, I think it's assuming the `adjust_nsize_for_ens` behaviour which was unintentional and has been removed.  Secondly, the "magic number" for the 6 km radius.  This should be a class member of `WindDirection`, constant if necessary, but with appropriate justification in docstrings or linked to elsewhere.

Recommend something resembling:
```
def __init__(...):
...
    self.nb_radius = 6000.   # comment here or in docstring
```
and use of this directly in the `NeighbourhoodProcessing` call.

Reply:
Suggestion implemented.